### ISSUE TITLE
Escaping typographer

### ISSUE THREAD

**Issue by TimothyGu** [markdown-it/markdown-it#271]
Currently I have this sentence:

```html
There should be two non-breaking spaces surrounded by
normal single quotes: <code>'&nbsp;&nbsp;'</code>.
```

(I'm using `<code>` since the backtick syntax seems to forcibly escape all characters in them.)

This then causes a problem with the typographer, as it transforms the code into the following HTML with curly quotes:

```html
There should be two non-breaking spaces surrounded by
normal single quotes: <code>’  ’</code>
```

I also tried to use `&apos;` with no luck:

```html
There should be two non-breaking spaces surrounded by
normal single quotes: <code>&apos;&nbsp;&nbsp;&apos;</code>.
```

The code above returns the same result as not using `&apos;`.

----

So my question is, is there a way to temporarily disable typography, or an escape syntax?

**Comment by puzrin** [comment]
Any html codes are passed as raw, without any analysis (pairing and so on). That's a CM spec and can't be changes. So, typografer does not know that that it's inside of code tag.

https://github.com/markdown-it/markdown-it/blob/master/lib/renderer.js#L21 may be override of default backticks renderer will help?

Please, clarify what is "temporary". You can change active rules for whole document only.

**Comment by TimothyGu** [comment]
"Temporary" as in "this specific code span."

I believe I may have found an additional bug/solution: jgm/commonmark.js#107

In fact, I can already use `cmark` to do what I want:

```sh
$ ./cmark --to html
`a  b` bleh `a  b`
<p><code>a b</code> bleh <code>a  b</code></p>
```

(the first code span is regular space, while the second uses non-breaking space)

**Comment by puzrin** [comment]
Let's summarize:

1. Collisions of typografer and raw html tags are not resolvable (may be kludges for specific cases can be done, but that's a wrong way in general to improve things)
2. Backticks are ok with typografer. If you have problems with backticks use, i can try to help with finding workaround. That's normal situation, and we did this parser specially to allow easy modifications.

Of cause, if CM spec is changed, this package will be updated quite fast.

Could you rephrase how can i help right now?

**Comment by TimothyGu** [comment]
Okay, I guess there isn't much I (or you) can do to fix this problem, and I now understand why. Thanks for the explanation.

**Comment by puzrin** [comment]
@TimothyGu after thinking a bit... you could clone backtick rule, replace markers with `<code>` / `<code/>` and make renderer non-escaping.
